src/Explorer/QueryCopilot/V2/Footer/Footer.tsx

```diff
@@ -29,18 +38,60 @@ export const Footer: React.FC = (): JSX.Element => {
   const handleEnterKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
     if (event.key === "Enter" && !event.shiftKey) {
       event.preventDefault();
-      handleSendMessage();
+      startSentMessageProcess();
     }
   };
 
-  const handleSendMessage = () => {
+  const handleSentQueryRequest = async (): Promise<void> => {
     if (userPrompt.trim() !== "") {
-      setChatMessages([...chatMessages, userPrompt]);
-      setUserPrompt("");
       setIsGeneratingQuery(true);
+      useTabs.getState().setIsTabExecuting(true);
+      useTabs.getState().setIsQueryErrorThrown(false);
+      setChatMessages([...chatMessages, { source: 0, message: userPrompt }]);
+      try {
+        if (shouldAllocateContainer) {
+          await explorer.allocateContainer();
```

Comment:
I don't think we should do any http request inside this component. If we have a client for calling the api let's handle all the request calls and handling, there instead of putting it in the component.

Reply:
Resolved